### ISSUE TITLE
Zoom fails VST2 Tracktion Linux and Windows

### ISSUE THREAD

**Issue by gitboyo** [surge-synthesizer/surge#512]
**Describe the bug**

FLStudio:
When resizing GUI via zoom, say to 150%, the GUI gets bigger but the container window does not so the GUI appears 'cut off'. Reloading reverts to 100% and problem goes away.

Tracktion Waveform:
When resizing GUI via zoom, say to 150%, the GUI gets bigger and rescales but the window title bar of the plugin vanishes. The plugin can then not be moved and can block the screen. Closing it and reopening if possible resets the plugin. If not possible (because plugin blocks whole screen) the project has to be closed.

**To Reproduce**

FLStudio:
1. Open instance of Surge.
2. Zoom from the menu to say, 150%

Tracktion Waveform:
1: Open instance of Surge.
2: Zoom from the menu to a larger size.

**Expected behavior**

FLStudio:
Plugin window of Surge resizes, but container window does not - cutting off the GUI.

Tracktion Waveform:
Plugin window loses ability to be moved due to title bar at top of window disappearing.

**Screenshots**

Surge Version:
![surge-1](https://user-images.githubusercontent.com/47363616/52293970-ff216c80-296f-11e9-8909-31c1842e53ce.PNG)

FLStudio Problem:
![surge-2](https://user-images.githubusercontent.com/47363616/52294018-16f8f080-2970-11e9-8d7f-3b2afb8802d0.PNG)

Inititial normal Tracktion Waveform Title bar of Surge:
![surge-3](https://user-images.githubusercontent.com/47363616/52294124-5a535f00-2970-11e9-897d-ae795fafac57.PNG)

Problem with Tracktion Waveform vanishing Title bar of Surge:
![surge-4](https://user-images.githubusercontent.com/47363616/52294168-76ef9700-2970-11e9-912f-5b782f80b820.PNG)

**Desktop (please complete the following information):**
 - OS:  windows 10
 - Host: FLStudio 20.1.1 64bit & Tracktion Waveform 8.2.7 64bit
 - Surge Version: VSTi v2.4 64bit - Build Feb 4  2019

**Comment by baconpaul** [comment]
Thank you.

As I'd mentioned in KVR, the collaboration between windows resizing and hosts is a bit tricky. It may be that FL calls the API which says "resize is impossible" and I am ignoring it, which would be at best a pyrrhic solution in that zoom would not be available but wouldn't work.

I can google of course, but are demo versions of these hosts available?

And appreciate the clear report; the book of work is pretty big. May be a bit before we turn to this. Just to make sure you don't end up disappointed. We will certainly look!

Thanks

**Comment by gitboyo** [comment]
Re: Demo Versions.

FLStudio has a very good demo available.


_Trial limitations

You can SAVE FL STUDIO projects.

REOPENING saved projects is NOT possible.

Some functions that use save/load capability don’t work in the trial. For example, "Clone channel" and the "Save preset as..." menu options.

You can EXPORT to WAV and MP3

The TRIAL is equivalent to FL Studio + All Plugin Bundle so you can test all features and plugins._


Download it here:
https://www.image-line.com/downloads/flstudiodownload.html

I'm positive you will be able to reproduce the bug here.


Tracktion Waveform also has a demo:
https://www.tracktion.com/products/waveform

Scroll to the bottom of the page to download (you need to sign up but it's free).

I'm not sure if you can download v8 as a demo, but it seems you can still buy it so it may be an option. Then again, there is a free version of Tracktion - v7 - and that may show the GUI bug as well. You can definitely get a demo there though so you should get something to work with.

Failing that, I don't think it would be unreasonable to ask for an NFR from their team. They seem like a good lot. They are fairly active on KVR, and it would be a good move for them to be seen to be helping you out.

To recap: 
Tracktion v7 is called 'T7' and is free. 
Tracktion v8 and v9 are now called 'Waveform' and they have a 30 day demo period.

Good luck and thanks for all your hard work.

**Comment by baconpaul** [comment]
Thank you both for the information and the thanks

My guess is if I look at this it will be week and a half or so - have a few other things ahead in queue. But will definitely look (unless other devs get to it ahead of me!)

**Comment by tank-trax** [comment]
I use Waveform9 and tested Surge in Linux and it does similar redrawing errors when resizing.

It seems to be an issue with Waveform. I've experienced similar issues with other plugins in this DAW.

For example with LinVST  (a wrapper for Windows VST's to be run in Linux hosts) there is a version especially for Waveform because of this behaviour...

> Tracktion/Waveform might have an embedded window offset problem where the plugin's gui is misaligned...

**Comment by baconpaul** [comment]
FLStudio 20.1.2 on windows 10 zoomed surge vst3 for me no problem, btw.

I propose we "fix" this issue by doing host/version detection and blacklisting some hosts from zoom. I don't see another path tbh.

**Comment by tank-trax** [comment]
using build 1.6.0-beta-8 - 2019-04-02 64 bit Linux in Waveform 9 on Debian Stretch

here are some examples of issues with resizing above 100%. It generally will open at 100% correctly 

![w9 open100](https://user-images.githubusercontent.com/41337033/56436485-0509c700-62aa-11e9-9fd3-fe898ff26fdf.png)

but tends to be inaccurate when resizing.

The behavior I am noticing though is hit and miss. Sometimes it will lock in properly and other times it will displace.

If using zoom down or Shift- it will miss the alignment, for example

![w9 100x 01](https://user-images.githubusercontent.com/41337033/56436773-fcfe5700-62aa-11e9-9219-debe0d9a1f3d.png)

Pressing Shift+ will grow it misaligned but after a few attempts it will grow into place and fit snugly. 

I am also noticing that in Waveform, if resized back down to 100% with an offset, it will fit back in if the synth is hidden and then re-displayed without having to delete the VST and then re-add it. 

This does not work on all dimensions though. It seems to be most stable at 100% and hit and miss with other sizes.

Here is an example of Zoom to 150% which offsets but then Zoom down by 10% produces a clean fit at 140%

![w9 150x](https://user-images.githubusercontent.com/41337033/56436621-7c3f5b00-62aa-11e9-83c4-fee31a369493.png)

**Comment by baconpaul** [comment]
With 1.6.1.1 FL20 zooms fine with the VST3 so renaming this to the Tracktion issue and linking to #938
